### PR TITLE
Move preload generation to Jenkins

### PR DESCRIPTION
**Pull request by ilya-zuyev**
This PR changes preload generator to run it in Jenkins environment

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -683,11 +683,17 @@ ifndef AUTOPUSH
 endif
 	env $(X_BUILD_ENV) docker buildx build --builder $(X_DOCKER_BUILDER) --platform $(KICBASE_ARCH) $(addprefix -t ,$(KICBASE_IMAGE_REGISTRIES)) --push  --build-arg COMMIT_SHA=${VERSION}-$(COMMIT) ./deploy/kicbase
 
+
```
Comment (medyagh) [inline Makefile:686]: comment, build a binary named upload-binary

**Review by medyagh — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -683,11 +683,18 @@ ifndef AUTOPUSH
 endif
 	env $(X_BUILD_ENV) docker buildx build --builder $(X_DOCKER_BUILDER) --platform $(KICBASE_ARCH) $(addprefix -t ,$(KICBASE_IMAGE_REGISTRIES)) --push  --build-arg COMMIT_SHA=${VERSION}-$(COMMIT) ./deploy/kicbase
 
+PRELOAD_TOOL = out/preload-tool
```
Comment (medyagh) [inline Makefile:686]: I suggest using the same pattern as others,

File: hack/preload-images/preload_images.go [hunk 3]
```diff
@@ -40,61 +42,106 @@ var (
...
 }
 
 func main() {
-	defer func() {
-		if err := deleteMinikube(); err != nil {
-			fmt.Printf("error cleaning up minikube: %v \n", err)
-		}
-	}()
+	flag.Parse()
+	viper.Set("preload", "true")
```
Comment (medyagh) [inline hack/preload-images/preload_images.go:63]: if not used at all, lets remove if used, we can keep it

File: hack/preload-images/preload_images.go [hunk 4]
```diff
@@ -40,61 +42,106 @@ var (
...
-			}
-			fmt.Printf("A preloaded tarball for k8s version %s - runtime %q doesn't exist, generating now...\n", kv, cr)
-			if err := generateTarball(kv, cr, tf); err != nil {
-				exit(fmt.Sprintf("generating tarball for k8s version %s with %s", kv, cr), err)
+			if *limit > 0 && i >= *limit {
+				break out
 			}
-			if err := uploadTarball(tf); err != nil {
-				exit(fmt.Sprintf("uploading tarball for k8s version %s with %s", kv, cr), err)
+			if *force || !download.PreloadExists(kv, cr) {
```
Comment (medyagh) [inline hack/preload-images/preload_images.go:87]: add a comment what this will do.

Comment (medyagh) [reply hack/preload-images/preload_images.go:87]: and also in the UI tell user, we are gonna overwrite...so we can see in the jenkins logs we are overwrtting

File: hack/preload-images/preload_images.go [hunk 5]
```diff
@@ -40,61 +42,106 @@ var (
...
+		k8sVersions = recent
+	}
+	return append([]string{
+		constants.DefaultKubernetesVersion,
+		constants.NewestKubernetesVersion,
+		constants.OldestKubernetesVersion,
+	}, k8sVersions...), nil
+}
+
+func process(cfg preloadCfg) error {
```
Comment (medyagh) [inline hack/preload-images/preload_images.go:121]: better name

**Review by medyagh — CHANGES_REQUESTED** [review]

**Review by medyagh — APPROVED** [review]

File: hack/preload-images/preload_images.go (hunk 3 above)
Comment (ilya-zuyev) [reply hack/preload-images/preload_images.go:65]: we need this flag for `download.PreloadExists` method to work

**Pull request merged**